### DISCUSSION TITLE
Problem with alpine x-if directive in livewire

### DISCUSSION THREAD

**Discussion by ShekhSaifuddin007** [alpinejs/alpine#883] [1. Help]
I'm just wondering that is a bug or else..! my problem is I'm using `x-if` directive to completely hide `livewire-component` from dom. but happen is that when using `x-if` any livewire function not work. but if I use `x-show` it works example is see my image bellow
![problem](https://user-images.githubusercontent.com/48278861/99672466-a6b45400-2a9d-11eb-92fe-098b779bb289.PNG)
 
I want to increment the table when I click the `plus` button beside of `Add Variation` text but it does not work, inside the image you see above the table I have a dropdown `Product Type` when I change the dropdown it gives me the different table when I go variable by changing the dropdown it will give this table whatever in the image, and when it will come I want to increment my table by clicking `plus` button but I can't. 😢😢, here I can use `x-show` but this not will suitable for my project any have any solution on this, please

I've created a [Laravel Playground](https://laravelplayground.com/#/snippets/909bb13c-12dc-445f-b2c9-54feb7df523d) on this see

### my blade

```blade
<form wire:submit.prevent="store">
   <div x-data="{ variation : 'single'}">
     <div class="row">
        <div class="col-lg-4">
           <div class="show-row">
                <fieldset class="form-group mb-0">
                   <div class="input-group">
                      <div class="input-group-append">
                           <label class="input-group-text" for="prtyp">Product Type *</label>
                      </div>
                      <select x-model="variation" wire:model.defer="product_type" class="custom-select" id="prtyp">
                          <option value="single">Single</option>
                          <option value="variable">Variable</option>
                      </select>
                   </div>
               </fieldset>
             </div>
          </div>
       </div>

        <template x-if="variation === 'single'" x-cloak>
            <livewire:products.variables.single/>
        </template>

        <template x-if="variation === 'variable'" x-cloak>
           <livewire:products.variables.variable/>
        </template>
    </div>
</form>
```

### my variable component

```php
<?php

 namespace App\Http\Livewire\Products\Variables;

 use Livewire\Component;

 class Variable extends Component
 {
  public $variationTables = [0];

 public function addMoreVariationTables()
 {
      $this->variationTables[] += 1;
 }

 public function render()
 {
      return view('livewire.products.variables.variable');
 }
}
```

### my variable component blade

```blade
<div class="row">
    <div class="col-12">
        <p>
            <span class="font-size-large font-weight-bolder">Add Variation</span>
            <button type="button" class="btn btn-primary btn-sm ml-1" wire:click.prevent="addMoreVariationTables">
                <i class="bx bx-plus-circle"></i>
            </button>
        </p>
    </div>

    <div class="col-12 mt-3">
        @foreach ($variationTables as $tableKey => $table)
            <div class="table-responsive" wire:key="variationTables_{{ $tableKey }}">
                <table class="table table-striped text-center table-primary">
                    <thead>
                      .....
                    </thead>
                    <tbody>
                    <tr>
                        <td class="p-1">
                            ....
                        </td>
                        <td>
                            <table class="table table-bordered text-center">
                                <thead>
                                    .....
                                </thead>
                                <tbody>
                                    .......
                                </tbody>
                            </table>
                        </td>
                    </tr>
                    </tbody>
                </table>
            </div>
        @endforeach
    </div>
</div>
```

**Comment by ryangjchandler** [comment]
Livewire won't register any event listeners, etc on the element since it's wrapped in an `x-if`, meaning it's not _really_ present in the DOM. When the `x-if` is `true` and the element is added in, it sounds like Livewire won't pick up that something has been added and won't attach them later on.

You're probably better off using `x-show`.

**Reply by ryangjchandler** [reply]
Looking at your component again though, you could also use a Blade `@if` instead of an Alpine `x-if`, since the `product_type` is being synced back to the server anyway.

**Reply by ShekhSaifuddin007** [reply]
Yes, `@if` could be a better solution, but I want this instant, with `wire:model` it little laggy and it sends the extra request to the server which I don't want, If there is a no other solution then we have to use like that.

**Reply by ryangjchandler** [reply]
`x-show` will have the same effect. You'll want to add `x-cloak` to the one that is hidden by default so that you don't get  a flash of content.
